sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -63,6 +64,12 @@
   private final List<Link> links;
   // Number of links recorded.
   private final int totalRecordedLinks;
+  // Max number of attibutes per span.
+  private final int maxNumberOfAttributes;
+  // Number of attributes recorded.
+  private int totalAttributeCount = 0;
```

Comment:
this should be moved down into the mutable section below and marked as `@GuardedBy("lock")`

Reply:
Done :+1: